### DISCUSSION TITLE
DataTemplate RadioButton CheckChanged event subscription with RxUI

### DISCUSSION THREAD

**Discussion by PhDave13** [reactiveui/ReactiveUI#2755] [Q&A]
In Xamarin Forms I have a Stacklayout, orientation horizontal which is databounded.

I have a BindableLaout.ItemTemplate that has a DataTemplate with 1 RadioButton.

As there could be 0 or many rendered Radio Buttons, how in ReactiveUI would I subscribe to the CheckChanged event of the radio buttons? I could set the group name for the Radio buttons but I still cannot see how I can subscribe to the CheckChanged events of the radio buttons.

I would like to subscribe the CheckChanged events so that I can perform a task dependent on the value of the content of the radio button selected.

```xml
         <StackLayout 
            x:Name="MyStackLayout"
            Orientation="Horizontal"
            RadioButtonGroup.GroupName="MyRadioButtonGroup"
            BindableLayout.ItemsSource="{Binding Source={x:Reference This}, Path=RadioButtonOptions}"
            >
                <BindableLayout.ItemTemplate>
                    <DataTemplate>
                        <RadioButton
                            HorizontalOptions="FillAndExpand"
                            Content="{Binding Path=OptionName}"
                            GroupName="MyRadioButtonGroupName"
                            IsChecked="{Binding IsChecked}" />
                    </DataTemplate>
                </BindableLayout.ItemTemplate>
            </StackLayout>
 ```

**Comment by glennawatson** [comment]
You could probably respond to the `IsChecked` value changing in your ViewModel and invoke the action there. You can use a `WhenAny` to respond.

**Reply by PhDave13** [reply]
Another twist, this in a Custom Control using BindableProperties.  So no VM.  Just thinking out loud here, but that shouldn't be a problem, I'll just add the WhenAny in the ctor of the Custom Control right?

Also, I will have several different groups of radio buttons on the page.  Is it possible to filter by the radio button group I wish to listen to?
